tests/test_validator.py

```diff
@@ -32,6 +36,48 @@ def set_validator(validator_name):
     os.environ["NBFORMAT_VALIDATOR"] = validator_name
 
 
+@pytest.mark.parametrize("validator_name", VALIDATORS)
+def test_should_warn(validator_name):
+    """Test that a v4 notebook witout id emit a warning"""
+    set_validator(validator_name)
+    with TestsBase.fopen("test4.5.ipynb", "r") as f:
+        nb = read(f, as_version=4)
+
+    del nb.cells[3]["id"]
+    assert nb.cells[3].get("id") is None
+    assert nb.cells[3]["cell_type"] == "code"
+
+    nb_copy = deepcopy(nb)
+
+    with pytest.warns(MissingIDFieldWarning):
+        validate(nb)
+    assert isvalid(nb) is True
+
+
+@pytest.mark.xfail(reason="In the future we want to stop warning, and raise an error")
+@pytest.mark.parametrize("validator_name", VALIDATORS)
+def test_should_not_mutate(validator_name):
+    """Test that a v4 notebook without id raise an error and does/not mutate
+
+    Probably should be 2 test. To enable in the future.
+    """
+    set_validator(validator_name)
+    with TestsBase.fopen("test4.5.ipynb", "r") as f:
+        nb = read(f, as_version=4)
+
+    del nb.cells[3]["id"]
+    assert nb.cells[3].get("id") is None
+    assert nb.cells[3]["cell_type"] == "code"
+
+    nb_deep_copy = deepcopy(nb)
+    with pytest.raises(MissingIDFieldWarning):
+        validate(nb)
+
+    assert nb == nb_deep_copy
+
+    assert isvalid(nb) is True
```

Comment:
if the call to `validate` raised, why would `isvalid` return True ?

Reply:
Yeah sorry I got confused here, the problem is that  `isvalid()` will currently return True, even if the notebook is not valid as.... it tries to fix it. 

This is currently marked as xfail. I'll fix it.